app/controllers/players/index.js

```diff
@@ -4,55 +4,125 @@ import { tracked } from '@glimmer/tracking';
 
 const ASCENDING_DIRECTION = 'ASC';
 const DESCENDING_DIRECTION = 'DESC';
+const DOWN_ARROW = '&darr;';
+const UP_ARROW = '&uarr;';
 
 export default class IndexController extends Controller {
-    get pageArray() {
-        let array = []
-
-        for (let index = 1; index <= Math.ceil(this.model.meta.count / this.limit); index++) {
-            array.push(index);
-        }
+    get pageValues() {
+        let array = new Array(Math.ceil(this.model.meta.count / this.limit)).fill().map((_, i) => i + 1);
 
         return array;
     }
 
     @tracked filterField;
+    @tracked filterFieldContent = this.filterField;
     @tracked filterValue;
+    @tracked filterValueContent = this.filterValue;
 
     @tracked sortField = 'id';
     @tracked sortDirection = ASCENDING_DIRECTION;
+    @tracked sortDirectionArrow = DOWN_ARROW;
 
     @tracked page = 1;
     @tracked limit = 5;
 
-    filterFieldValues = [
-        'firstName',
-        'lastName'
+    selectOptions = [
+        {
+            text: 'First name',
+            value: 'firstName'
+        },
+        {
+            text: 'Last name',
+            value: 'lastName'
+        }
+    ]
+
+    tableOptions = [
+        {
+            text: 'ID',
+            value: 'id'
+        },
+        {
+            text: 'First name',
+            value: 'firstName'
+        },
+        {
+            text: 'Last name',
+            value: 'lastName'
+        }
     ]
 
     limitValues = [5, 10, 20, 50];
 
+    resetPage() {
+        this.page = 1;
+    }
+
+    movePage(amount) {
+        this.page += amount;
+    }
+
     @action
-    selectFilterField(filterField) {
-        this.filterField = filterField;
+    choseFilterField(selectedObject) {
+        this.filterFieldContent = selectedObject.value;
     }
 
     @action
     onFilterClick() {
-        this.filterValue = this.value;
+        if (this.filterValueContent === '') {
+            this.filterValue = undefined;
+            return;
+        }
 
-        this.page = 1;
+        this.filterField = this.filterFieldContent;
+        this.filterValue = this.filterValueContent;
+        this.resetPage();
+    }
+
+    @action
+    onFilterResetClick() {
+        this.filterField = undefined;
```

Comment:
In general undefined represents a value that is not defined yet. Might be it makes sense to set null instead of the undefined.